rcl/src/rcl/arguments.c

```diff
@@ -354,6 +354,47 @@ rcl_get_zero_initialized_arguments(void)
   return default_arguments;
 }
 
+rcl_ret_t
+rcl_remove_ros_arguments(
+  char const * const argv[],
+  rcl_arguments_t * args,
+  rcl_allocator_t allocator,
+  int * nonros_argc,
+  const char ** nonros_argv[])
+{
+  RCL_CHECK_ALLOCATOR_WITH_MSG(&allocator, "invalid allocator", return RCL_RET_INVALID_ARGUMENT);
+  RCL_CHECK_ARGUMENT_FOR_NULL(argv, RCL_RET_INVALID_ARGUMENT, allocator);
+  RCL_CHECK_ARGUMENT_FOR_NULL(nonros_argc, RCL_RET_INVALID_ARGUMENT, allocator);
+  RCL_CHECK_ARGUMENT_FOR_NULL(args, RCL_RET_INVALID_ARGUMENT, allocator);
+
+  *nonros_argc = rcl_arguments_get_count_unparsed(args);
+  *nonros_argv = NULL;
+
+  if (*nonros_argc > 0) {
```

Comment:
`nonros_argc` could be `-1` if `args` is zero-intialized (it would get past the `NULL` check above). Maybe this should return `RCL_RET_INVALID_ARGUMENT` in that case?

Reply:
Maybe I missed why, but should `nonros_argc` just be unsigned, i.e. `size_t`?

Reply:
I treated it as signed because the `main` signature is also signed

Reply:
Ah, you're right, never mind.

Reply:
`rcl_arguments_get_count_unparsed` returns `-1` when args is `NULL` or zero initialized. If that's undesirable a couple alternatives are returning an unsigned value that is `0` if the arguments are invalid, or returning `rcl_ret_t` and using an output parameter.

Reply:
Nah, unless @mjcarroll thinks so, I'd say leave it and handle the negative case.